src/views/instances/list/useDeployments.tsx

```diff
@@ -0,0 +1,42 @@
+import { useAuthAxios } from '../../../hooks'
+import { GroupsWithDeployments } from '../../../types'
+import { useState, useMemo } from 'react'
+
+const CURRENT_USER_LOCAL_STORAGE_KEY = "DHIS2_IM_CURRENT_USER";
+
+const useDeployments = () => {
+    const [{ data, error, loading }, refetch] = useAuthAxios<GroupsWithDeployments[]>('/deployments', {
+        useCache: false,
+    })
+    const [showOnlyMyInstances, setShowOnlyMyInstances] = useState(false)
+
+    const getCurrentUserFromLocalStorage = () => {
+        const user = localStorage.getItem(CURRENT_USER_LOCAL_STORAGE_KEY);
+        return user ? JSON.parse(user) : null;
+    }
+
+    const currentUser = useMemo(() => getCurrentUserFromLocalStorage(), []);
+
+    const filterDeploymentsByUser = useMemo(() => {
+        if (!currentUser) return data;
+
+        return data?.map(group => ({
+            ...group,
+            deployments: group.deployments.filter(deployment => deployment.user.id === currentUser.id)
+        }))
+
+    }, [data, currentUser]);
```

Comment:
Yes, this should definitely be done server side! Which is also mentioned on the Jira [task](https://dhis2.atlassian.net/jira/core/projects/DEVOPS/board?selectedIssue=DEVOPS-418).

In regards to the alternative solution. The original design specs did include a "Created by me" page and I'm not against introducing such.

I've created [this](https://dhis2.atlassian.net/browse/DEVOPS-425) Jira task so there's a greater chance this will get prioritized.

Reply:
Alright, I'll resolve this comment